cfgov/ask_cfpb/models/django.py

```diff
@@ -0,0 +1,321 @@
+        FieldPanel('snippet', classname="full"),
+        FieldPanel('answer', classname="full"),
+        FieldPanel('slug'),
+        FieldPanel('question_es', classname="title"),
+        FieldPanel('snippet_es', classname="full"),
+        FieldPanel('answer_es', classname="full"),
+        FieldPanel('slug_es'),
+        FieldPanel('audiences', widget=forms.CheckboxSelectMultiple),
+        FieldPanel('next_step'),
+        FieldPanel('category', widget=forms.CheckboxSelectMultiple),
+        FieldPanel(
+            'subcategory', widget=forms.CheckboxSelectMultiple),
+        FieldPanel('related_questions', widget=forms.SelectMultiple),
+        FieldPanel('tagging')
+    ]
+
+    def answer_text(self):
+        """Unescapes and removes html tags from answer fields"""
+        unescaped = ("{} {}".format(
+            html_parser.unescape(self.snippet),
+            html_parser.unescape(self.answer)))
+        return html.strip_tags(unescaped).strip()
+
+    def answer_text_es(self):
+        """Unescapes and removes html tags from Spanish answer fields"""
+        unescaped = ("{} {}".format(
+            html_parser.unescape(self.snippet_es),
+            html_parser.unescape(self.answer_es)))
+        return html.strip_tags(unescaped).strip()
+
+    def cleaned_questions(self):
+        cleaned_terms = html_parser.unescape(self.question)
+        return [html.strip_tags(cleaned_terms).strip()]
+
+    def cleaned_questions_es(self):
+        cleaned_terms = html_parser.unescape(self.question_es)
+        return [html.strip_tags(cleaned_terms).strip()]
+
+    def subcat_slugs(self):
+        cats = [cat.slug for cat in self.subcategory.all()]
+        return cats
+
+    def category_text(self):
+        if self.category.all():
+            return [cat.name for cat in self.category.all()]
+        else:
+            return ''
+
+    def audience_strings(self):
+        for audience in self.audiences.all():
+            yield audience.name
+
+    def tags(self):
+        for tag in self.tagging.split(','):
+            tag = tag.replace('"', '')
+            tag = tag.strip()
+            if tag != u'':
+                yield tag
+
+    def create_or_update_page(self, language=None):
+        """Create or update an English or Spanish Answer page"""
+        answer_parent = Page.objects.get(slug=PARENT_SLUG)
+        if language == 'en':
+            _slug = self.slug
+            _question = self.question
+            _snippet = self.snippet
+            _answer = self.answer
+        elif language == 'es':
+            _slug = self.slug_es
+            _question = self.question_es
+            _snippet = self.snippet_es
+            _answer = self.answer_es
+        else:
+            raise Exception('unsupported language: "{}"'.format(language))
+        base_page = get_or_create_page(
+            apps,
+            'ask_cfpb',
+            'AnswerPage',
+            '{}: {}'.format(self.id, _question[:248]),
```

Comment:
You might want to find out if the stakeholders actually care about the "self healing" functionality or having the numbers in the URL. If not, then this can still be the basis for catching those URL's and redirecting to the new ones